lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
A follow-up on the bridgedGepIterator.
I had another look at the test case in your link "https://github.com/jcarlson23/bridgedGEPiterator"

I found the root cause. I believe the type of "*gi" (llvm-3.8.0) is one level higher than that of the iterator "gi.getIndexedType()" (llvm-4.0). 

Please see below.

$ ./bin/testgep ../simple.bc 
  %1 = alloca %struct.MyStruct*, align 8
  %2 = load %struct.MyStruct*, %struct.MyStruct** %1, align 8
  %3 = getelementptr inbounds %struct.MyStruct, %struct.MyStruct* %2, i32 0, i32 0

Use the 4.0 IndexedType()
%struct.MyStruct = type { i32, i8* }
i32 0
i32
i32 0

**Current bridgedGepIterator implementation:**
%struct.MyStruct*
i32 0
%struct.MyStruct*
i32 0


**The ideal implementation (same as that of llvm-3.8.0) should be:**
%struct.MyStruct*
i32 0
%struct.MyStruct
i32 0

Thanks